src/orders/components/OrderRefundReturnAmount/OrderRefundReturnAmount.tsx

```diff
@@ -121,18 +121,42 @@ const OrderRefundAmount: React.FC<OrderRefundAmountProps> = props => {
     replacedProductsValue
   } = amountData;
 
-  const selectedRefundAmount =
+  const isRefundAutomatic =
     type === OrderRefundType.PRODUCTS &&
-    data.amountCalculationMode === OrderRefundAmountCalculationMode.AUTOMATIC
-      ? refundTotalAmount?.amount
-      : data.amount;
+    data.amountCalculationMode === OrderRefundAmountCalculationMode.AUTOMATIC;
+
+  const selectedRefundAmount = isRefundAutomatic
+    ? refundTotalAmount?.amount
+    : data.amount;
 
   const isAmountTooSmall = selectedRefundAmount && selectedRefundAmount <= 0;
   const isAmountTooBig = selectedRefundAmount > maxRefund?.amount;
 
-  const disableRefundButton = isReturn
-    ? disableSubmitButton || isAmountTooSmall || isAmountTooBig
-    : !selectedRefundAmount || isAmountTooBig || isAmountTooSmall;
+  const parsedRefundTotalAmount = isAmountTooBig
+    ? maxRefund
+    : refundTotalAmount;
+
+  const shouldDisableRefundButton = () => {
```

Comment:
```suggestion
  const shouldRefundButtonBeDisabled = () => {
```
I think it's a bit clearer